### PR TITLE
Fixed NavLink Issue 

### PR DESCRIPTION
**Pull request by bilalk711**
### Original Issue 
[NavLink active breaks when to contains trailing slash](https://github.com/remix-run/react-router/issues/10506)

### Steps to Reproduce
Create a NavLink with end omitted or set to false and with to set to a route path which contains sub-routes and ensure a trailing slash at the end.
Visit a sub-route of the path in use.

### Expected Behaviour
With end false, when the current location is a sub-route of a mounted NavLink, then it should be active.

### Actual Behaviour
NavLink is considered inactive.

This issue is fixed in this PR.

### CODE REVIEW

**Comment by timdorr** [comment]
Just as an FYI, your commits aren't attributed to your GitHub account because you don't have your Git email address added. If you add bilal@dubicars.com to GitHub, you'll get attribution to those commits and the CLA bot will be happy :)

**Comment by bilalk711** [comment]
> Just as an FYI, your commits aren't attributed to your GitHub account because you don't have your Git email address added. If you add [bilal@dubicars.com](mailto:bilal@dubicars.com) to GitHub, you'll get attribution to those commits and the CLA bot will be happy :)

**Comment by bilalk711** [comment]
> Just as an FYI, your commits aren't attributed to your GitHub account because you don't have your Git email address added. If you add [bilal@dubicars.com](mailto:bilal@dubicars.com) to GitHub, you'll get attribution to those commits and the CLA bot will be happy :)

Actually the issue was that I changed my username but forgot to add that to contributors.yml

File: packages/react-router-dom/__tests__/nav-link-active-test.tsx [hunk 1]
```diff
@@ -397,6 +397,33 @@ describe("NavLink", () => {
       expect(anchor.props.className).toMatch("active");
     });
```
Comment (iamdey) [inline packages/react-router-dom/__tests__/nav-link-active-test.tsx:399]: you can eventually add a test to make sure a «Root» Navlink is not always active:


```
<NavLink to="/">Home</NavLink>
…
let anchor = renderer.root.findByType("a");
expect(anchor.props.className).not.toMatch("active");
```

**Review by iamdey — COMMENTED** [review]
It's a bit out-of-scope but it would be nice to change the documentation to make it consistent with reality:

https://reactrouter.com/en/main/components/nav-link#end

```diff
Without the end prop, this link is always active because every URL matches /.

- <NavLink to="/">Home</NavLink>
+ <NavLink to="/page">Page</NavLink> 

+ Exception for `/` which will be active only when url match path exactly

To match the URL "to the end" of to, use end:

- <NavLink to="/" end>
+ <NavLink to="/page" end>
  Home
</NavLink>





Link | URL | isActive
-- | -- | --
<NavLink to="/tasks" /> | /tasks | true
<NavLink to="/tasks" /> | /tasks/123 | true
+ <NavLink to="/tasks/" /> | /tasks | true
+ <NavLink to="/tasks/" /> | /tasks/123 | true
<NavLink to="/tasks" end /> | /tasks | true
<NavLink to="/tasks" end /> | /tasks/123 | false
+ <NavLink to="/tasks/" end /> | /tasks | false
+ <NavLink to="/tasks/" end /> | /tasks/ | true
+ <NavLink to="/tasks/" end /> | /tasks/123 | false

```

File: packages/react-router-dom/__tests__/nav-link-active-test.tsx (hunk 1 above)
Comment (bilalk711) [reply packages/react-router-dom/__tests__/nav-link-active-test.tsx:399]: I will add the test for this PR but for documentation I will have to create another one.
Thanks for your review!!
Really appreciate that.

Comment (bilalk711) [reply packages/react-router-dom/__tests__/nav-link-active-test.tsx:399]: Added that test `does not apply the default 'active' className to the underlying <a> when at root'/'`

**Comment by bilalk711** [comment]
@iamdey can you merge it now if you have write access?

**Comment by bilalk711** [comment]
Updated the documentation as per new changes.

**Comment by bilalk711** [comment]
@brophdawg11 Do you have write access or can you mention someone with authority who can merge this PR?

File: packages/react-router-dom/__tests__/nav-link-active-test.tsx [hunk 2]
```diff
@@ -397,6 +397,57 @@ describe("NavLink", () => {
...
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      let anchor = renderer.root.findByType("a");
+      expect(anchor.props.className).toMatch("active");
+    });
+    
+    it("does not apply the default 'active' className to the underlying <a> when at root'/'", ()=>{
```
Comment (brophdawg11) [inline packages/react-router-dom/__tests__/nav-link-active-test.tsx:426]: Can you elaborate on what this test is for?  This doesn't seem like it's testing logic related to a trailing slash on child routes?

File: packages/react-router-dom/index.tsx [hunk 3]
```diff
@@ -657,12 +657,13 @@ export const NavLink = React.forwardRef<HTMLAnchorElement, NavLinkProps>(
         : null;
       toPathname = toPathname.toLowerCase();
     }
-
+    const endSlashPosition = (toPathname.length > 1 && toPathname[toPathname.length - 1] === "/") ? toPathname.length - 1 : toPathname.length;
```
Comment (brophdawg11) [inline packages/react-router-dom/index.tsx:660]: We can use `endsWith` here for a little more clarity:

```suggestion
    const endSlashPosition =
      toPathname.length > 1 && toPathname.endsWith("/")
        ? toPathname.length - 1
        : toPathname.length;
```

**Pull request merged**